docs/extending/overview_extending.rst

```diff
@@ -0,0 +1,201 @@
+   * - XBlock (``xblock.v1``)
+     - Adopt, Stable
+     - An XBlock provides a new type of learnable content that can be used in courses, content libraries, etc. See `Content Integrations`_.
+   * - XBlock unit tests (``xblock.test.v0``)
+     - Assess, Limited
+     - XBlocks can also install test code that will then be run alongside the platform's usual python unit tests. It's unclear how well-supported this is at the moment.
+
+.. _Application: https://docs.djangoproject.com/en/3.0/ref/applications/
+.. _Django app plugin documentation: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/plugins/README.rst
+.. _Plugin Contexts: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/plugins/docs/decisions/0003-plugin-contexts.rst
+.. _course tabs documentation: https://openedx.atlassian.net/wiki/spaces/AC/pages/30965919/Adding+a+new+course+tab
+.. |course_tools.py| replace:: ``course_tools.py``
+.. _course_tools.py: https://github.com/edx/edx-platform/blob/master/openedx/features/course_experience/course_tools.py
+.. _Adding Custom Fields to the Registration Page: https://edx.readthedocs.io/projects/edx-installing-configuring-and-running/en/latest/configuration/customize_registration_page.html
+.. |learning_context.py| replace:: ``learning_context.py``
+.. _learning_context.py: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/xblock/learning_context/learning_context.py
+.. |UserPartition docstring| replace:: ``UserPartition`` docstring
+.. _UserPartition docstring: https://github.com/edx/edx-platform/blob/f8cc58618a39c9f7b8e9e1001eb2d7a10395797e/common/lib/xmodule/xmodule/partitions/partitions.py#L105-L120
+
+**********************
+Platform look and feel
+**********************
+
+
+Themes ("comprehensive theming")
+================================
+
+*Status: Hold, Stable*
+
+Changing the look and feel of the edX platform is generally done by creating a new "theme". See `Changing Themes for an Open edX Site`_ for documentation. Note that most theming documentation applies to the legacy UI components used in edX, which are .html files (django/mako templates) rendered by the backend and styled using either the "v1" or "v2" (a.k.a. "Pattern Library") stylesheets. However, the platform UI is slowly being replaced by new React-based "micro-frontends" (MFEs), and a different approach is required for theming MFEs (see `Theming Micro-frontends`_).
+
+Theming micro-frontends
+=======================
+
+*Status: Trial, Limited*
+
+Methods for theming MFEs are still being developed. It is likely to involve:
+
+#. Branding: modifying fonts, colors, and logos via themes/css (there is an |example edx theme|_ that you can use as a template for defining fonts and colors, but some MFEs currently lack a mechanism for changing the theme).
```

Comment:
Is "theming micro-frontends" still considered to be in "trial, limited" status? There's a whole topic on micro-frontends in the "new" developer documentation that's publicly available right now.

I would suggest moving some or all of this content to the micro-frontends topic and linking to that topic from within this section.